### DISCUSSION TITLE
Custom React Environment not working on Bit.dev

### DISCUSSION THREAD

**Discussion by lhassis** [teambit/bit#5014] [Help] [answered]
Hi everyone!

I've create a custom react environment to provide Theme to my components preview. When I run bit server locally (bit start command) the components render correctly, but after publishing the environment and the component to bit.dev server, it fails to render, since it does not have the theme options.
Is it possible to use this custom environment on bit.dev? If so, what configuration steps am I missing?

Thanks in advance!
Regards,
Lucas Assis

**Edit:**
To better clarify the problem, I've created a public scope on bit.dev and published my theme provider and a button component, that depends on theme. It can be found at: https://bit.dev/lhassis/theme-test

Also, I have pushed on Github the source code for this workspace.
https://github.com/lhassis/bit-theme-preview-test

**Comment by sbland** [comment]
Have you also published your custom environment? Can you access it's docs on bit.dev?

**Reply by lhassis** [reply]
Yes, I have.
It fails not only on bit.dev, but also on a local bit container.

**Reply by lhassis** [reply]
To better clarify the problem, I've created a public scope on bit.dev and published my theme provider and a button component, that depends on theme. It can be found at https://bit.dev/lhassis/theme-test

Also, I have pushed on Github the source code for this workspace.
https://github.com/lhassis/bit-theme-preview-test

I would appreciate any help.
Regards,
Lucas

**Comment by lhassis** [comment, marked as answer]
Solved after following: https://github.com/teambit/bit/discussions/5225
